src/DataMapper/ProductDataMapper.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusFacebookPlugin\DataMapper;
+
+use FacebookAds\Object\ServerSide\Content;
+use Setono\SyliusFacebookPlugin\ServerSide\ServerSideEventInterface;
+use Sylius\Component\Product\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+/* not final */ class ProductDataMapper implements DataMapperInterface
+{
+    /**
+     * @psalm-assert-if-true ProductInterface $source
+     */
+    public function supports($source, ServerSideEventInterface $target, array $context = []): bool
+    {
+        return $source instanceof ProductInterface;
+    }
+
+    /**
+     * @param ProductInterface|object $source
+     */
+    public function map($source, ServerSideEventInterface $target, array $context = []): void
+    {
+        Assert::isInstanceOf($source, ProductInterface::class);
+
+        $customData = $target->getCustomData();
+
+        /** @psalm-suppress PossiblyNullArgument */
+        $customData->setContentName($source->getName());
```

Comment:
Instead of suppressing you could just cast it to string

Reply:
In fact, Facebook accepts `null`s (it just has improper docblocks) and it's better to pass null rather than an empty string.